File: JavaNgArch/src/javangarch/mvc/shared/AppBaseController.java
package javangarch.mvc.shared;


import java.io.IOException;
import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;
import java.nio.file.DirectoryStream;
import java.nio.file.Files;
import java.nio.file.NoSuchFileException;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.function.Function;
import java.util.stream.Stream;

import javax.annotation.Resource;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.apache.commons.lang3.StringUtils;
import org.springframework.web.multipart.MultipartFile;



import projectbase.domain.ICommonBD;
import projectbase.mvc.AjaxHelperExtension;
import projectbase.mvc.AuthFailureException;
import projectbase.mvc.BaseController;
import projectbase.mvc.DisplayExtension;
import projectbase.mvc.ExcludeController;
import projectbase.mvc.WebConfigurationManager;
import projectbase.mvc.result.ActionResult;
import projectbase.mvc.result.HttpStatusCodeResult;
import projectbase.mvc.result.RichClientJsonResult;
import javangarch.domain.bdinterface.IAdminBD;

import javangarch.domain.domainmodel.gn.Corp;
import javangarch.domain.domainmodel.gn.User;
import javangarch.mvc.home.LoginInfoViewModel;

	@ExcludeController
    public class AppBaseController extends BaseController
    {
        @Resource public IAdminBD AdminBD;
        public static String DefaultPassword = "c4ca4238a0b923820dcc509a6f75849b";//默认密码为1
        public static final String AdminCode = "Admin"; //用于管理部门和用户的固定代码
        private static List<String> _relatedFileNames; 

 	 
       protected int GetLoginCorpId()
        {
            Corp corp = ((LoginInfoViewModel)getRequest().getSession().getAttribute("LoginInfo")).getLoginCorp();
            return corp.getId();
        }
       protected Corp GetLoginCorp()
       {
    	   Corp corp = ((LoginInfoViewModel) getRequest().getSession().getAttribute("LoginInfo")).getLoginCorp();
           AdminBD.RefreshCorp(corp);
           return corp;
       }
       protected User GetLoginUser()
       {
           User user = ((LoginInfoViewModel) getRequest().getSession().getAttribute("LoginInfo")).getLoginUser();
           AdminBD.RefreshUser(user);
           return user;
       }
 
        protected boolean CanAccess(String funcCode)
        {
        	LoginInfoViewModel logininfo = ((LoginInfoViewModel)getRequest().getSession().getAttribute("LoginInfo"));
            return logininfo.CanAccess(funcCode);
        }

  
        protected void CheckFuncAuth(String funcCode) 
        {
        	LoginInfoViewModel loginInfo = ((LoginInfoViewModel)getRequest().getSession().getAttribute("LoginInfo"));
            if (loginInfo.CanAccess(funcCode)) 
            	return;
            throw new AuthFailureException();
        }
  
        //登录用户是否为admin
        public boolean IsAdmin()
        {
            return IsAdmin(GetLoginUser());
        }

        //给定用户是否为admin
        public boolean IsAdmin(User user)
        {
            return user!=null && !user.IsTransient() && StringUtils.equalsIgnoreCase(user.getCode(),AdminCode);
        }

        protected ActionResult FileNotFound()
        {
         	return RcJsonError( "FileNotFound", "FileNotFound");
        }
       	@Override
        protected ActionResult OnException(HttpServletRequest req,
    			HttpServletResponse response, Object handler, Exception ex)
        {
            if (ex instanceof AuthFailureException)
            {
                return AuthFailure();
            }
            return super.OnException(req,response,handler,ex);
        }
 

    }

File: NetNgArch/src/NetNgArch.Web.Mvc/Shared/AppBaseController.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Web;
using System.Web.Configuration;
using System.Linq;
using System.Web.Mvc;
using NetNgArch.Domain.BDInterface;
using NetNgArch.Domain.DomainModel.GN;
using NetNgArch.Web.Mvc.Home;
using ProjectBase.Utils;
using ProjectBase.Web.Mvc;
using ProjectBase.Web.Mvc.Angular;


namespace NetNgArch.Web.Mvc.Shared
{
    public class AppBaseController : BaseController
    {
        public IAdminBD AdminBD { get; set; }
        public static string DefaultPassword = "c4ca4238a0b923820dcc509a6f75849b";//默认密码为1
        public static string AdminCode = "Admin"; //用于管理部门和用户的固定代码
  
        protected int GetLoginCorpId()
        {
            var corp = ((LoginInfoViewModel)Session["LoginInfo"]).LoginCorp;
            return corp.Id;
        }
       protected Corp GetLoginCorp()
       {
           var corp = ((LoginInfoViewModel) Session["LoginInfo"]).LoginCorp;
           AdminBD.RefreshCorp(corp);
           return corp;
       }
       protected User GetLoginUser()
       {
           var user = ((LoginInfoViewModel) Session["LoginInfo"]).LoginUser;
           AdminBD.RefreshUser(user);
           return user;
       }
      
        protected bool CanAccess(string funcCode)
        {
            var logininfo = (LoginInfoViewModel)Session["LoginInfo"];
            return logininfo.CanAccess(funcCode);

        }
 
        protected void CheckFuncAuth(string funcCode)
        {
            var logininfo = (LoginInfoViewModel)Session["LoginInfo"];
            if (logininfo.CanAccess(funcCode)) return;
            throw new AuthFailureException();
        }

        //登录用户是否为admin
        public bool IsAdmin()
        {
            return IsAdmin(GetLoginUser());
        }
        //给定用户是否为admin
        public bool IsAdmin(User user)
        {
            return user != null && !user.IsTransient() && user.Code.ToLower()==AdminCode.ToLower();
        }
        protected ActionResult FileNotFound()
        {
            return RcJsonError("FileNotFound", "FileNotFound");
        }
        protected override void OnException(ExceptionContext filterContext)
        {
            if (filterContext.Exception is AuthFailureException)
            {
                filterContext.Result = AuthFailure();
                filterContext.ExceptionHandled = true;
                return;
            }
            base.OnException(filterContext);
        }
        private string ToHtmlEntityString(string s)
        {
            var chars = s.ToCharArray();
            var entitystring = "";
            Array.ForEach(chars, c =>
                                     {
                                         var ss = HttpUtility.UrlEncodeUnicode(c + "").Replace("%u", "&#x");
                                         entitystring = entitystring + ss;
                                         if (ss.StartsWith("&#x")) entitystring = entitystring + ";";
                                     });
            return entitystring;
        }
       
    }
}
